uniffi/src/templates/ObjectTemplate.rs

```diff
@@ -50,14 +50,14 @@ lazy_static::lazy_static! {
         let mut err: ffi_support::ExternError = Default::default(); // XXX TODO: error handling!
         // If the method does not have the same signature as declared in the IDL, then
         // this attempt to call it will fail with a (somewhat) helpful compiler error.
-        UNIFFI_HANDLE_MAP_{{ obj.name()|upper }}.call_with_output_mut(&mut err, {{ meth.first_argument().name() }}, |obj| {
+        UNIFFI_HANDLE_MAP_{{ obj.name()|upper }}.{% match meth.error() %}{% when Some with (e) %}call_with_result_mut{% else %}call_with_output_mut{% endmatch %}(&mut err, {{ meth.first_argument().name() }}, |obj| {% match meth.error() %}{% when Some with (e) %}-> Result<{% match meth.ffi_func().return_type() %}{% when Some with (return_type) %}{{ return_type|ret_type_c }}{% else %}(){% endmatch %}>{% else %}{% endmatch %}{
             let _retval = {{ obj.name() }}::{{ meth.name() }}(
                 obj,
                 {%- for arg in meth.arguments() %}
                 {{ arg.name()|lift_rs(arg.type_()) }},
                 {%- endfor %}
-            );
-            {% match meth.return_type() %}{% when Some with (return_type) %}{{ "_retval"|lower_rs(return_type) }}{% else %}{% endmatch %}
+            ){% match meth.error() %}{% when Some with (e) %}?{% else %}{% endmatch %};
+            {% match meth.error() %}{% when Some with (e) %}Ok({% else %}{% endmatch %}{% match meth.return_type() %}{% when Some with (return_type) %}{{ "_retval"|lower_rs(return_type) }}{% else %}_retval{% endmatch %}{% match meth.error() %}{% when Some with (e) %}){% else %}{% endmatch%}
```

Comment:
Lots of repetitive `match` here, I'll look if there is a way to set a custom reusable block with askama

Reply:
Realized #21 adds exactly the kind of stuff that'll help here 👀